x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/read/SingletonOrdinalsBuilder.java

```diff
@@ -64,6 +65,41 @@ public SingletonOrdinalsBuilder endPositionEntry() {
         throw new UnsupportedOperationException("should only have one value per doc");
     }
 
+    BytesRefBlock tryBuildConstantBlock() {
+        if (minOrd == maxOrd) {
+            boolean seenNulls = false;
+            for (int ord : ords) {
+                if (ord == -1) {
+                    seenNulls = true;
+                    break;
+                }
+            }
+            if (seenNulls == false) {
+                final BytesRef v;
+                try {
+                    v = BytesRef.deepCopyOf(docValues.lookupOrd(minOrd));
+                } catch (IOException e) {
+                    throw new UncheckedIOException("failed to lookup ordinals", e);
+                }
+                BytesRefVector bytes = null;
+                IntVector ordinals = null;
+                boolean success = false;
+                try {
+                    bytes = blockFactory.newConstantBytesRefVector(v, 1);
+                    ordinals = blockFactory.newConstantIntVector(0, ords.length);
+                    final var result = new OrdinalBytesRefBlock(ordinals.asBlock(), bytes);
```

Comment:
Here, we choose to return an ordinal constant block instead of a direct constant block to ensure that ordinal optimizations are applied if the constant optimization is not available.